[ci] :racehorse: rake ci:run by splitting it

degemer opened the pull request:
`rake ci:run` is really slow (6min on Travis, but up to 15min locally), so nobody runs it before pushing.

This PR splits the default task into 3 tasks:
- `checks_mock` which runs all mocked tests for checks,
- `core_integration` which runs all integration tests of the agent core, (which takes a lot of time locally because of external calls & `time.sleep`)
- and `default`, which only runs core unit tests and `pylint` on changed files.

Pros:
* it takes 15s to run `rake ci:run` locally.
* nicer separation of tasks ?

Cons:
* Probably some overhead on Travis.

It has only been tested locally, so let's see what happens on Travis.

LeoCavaille commented on ci/default.rb:
```diff
@@ -40,11 +40,24 @@
+      }.select{ |stat_path|
+        status = stat_path.first
+        path = stat_path.last
+        path.end_with? ".py" and status != 'D'
```
It's a good idea but I think it should end up in a different rake task, people have different workflows. For instance in my case `git status --porcelain` would probably return nothing in 99% cases because my changes are always committed in a WIP commit on a branch and I carry non-committed files around.
Can you do maybe a `rake lint:changed` with this behavior and keep the `rake:lint`?

degemer replied:
I'm not sure to understand: if `git status --porcelain` returns nothing (or there is no `*.py` files, the old `rake:lint` task will run (`if paths.empty?`). (see for instance on Travis)
I really wanted to have this behavior to have a fast `rake ci:run` when working on something (ie right now, it takes ~15s when a `.py` file is changed, versus more than 2min because of pylint (which is is really slooooow)).

LeoCavaille commented on tests/core/test_common.py:
```diff
@@ -300,6 +300,3 @@ def test_dupe_tags(self):
         self.assertEquals(metric.value, 2)
-
-if __name__ == '__main__':
-    unittest.main()
```
nice cleanup!

LeoCavaille commented on tests/core/test_dogstatsd.py:
```diff
@@ -1,12 +1,17 @@
-# -*- coding: utf-8 -*-
+ # -*- coding: utf-8 -*-
```
nitpicky: leading space?

LeoCavaille commented on tests/core/test_dogstatsd.py:
```diff
@@ -1,12 +1,17 @@
 import time
 
+# 3p
 import unittest
```
This is stdlib

LeoCavaille commented on tests/core/test_flare.py:
```diff
@@ -1,6 +1,10 @@
+# stdlib
 import os.path
 import mock
+from nose.plugins.attrib import attr
```
You're missing 3p

LeoCavaille commented on tests/core/test_watchdog.py:
```diff
@@ -1,16 +1,23 @@
+import subprocess
+import sys
 import time
+import psutil
```
psutil is 3p

degemer commented:
Thanks for the review @LeoCavaille !
I updated my code as per your comments (except `rake lint`), and since it's a :sunny: day, I also removed all the `unitest.main()`. :tada:

degemer commented:
I `rubocop -a`'d the new `ci/` files.

degemer commented:
I changed `lint`: it is still the same old command by default, but I added a `lint:changed`.
